handbook/data-stores/mongodb.md

```diff
@@ -0,0 +1,138 @@
+  
+1. With a user with administrator privileges, log into the desktop of the SystemLink app server.
+1. Open **NI SystemLink Server Configuration**.
+1. Navigate to **NoSqlDatabase**.
+1. Under **The NoSqlServer is launched and managed by SystemLink**, adjust the value in **The cache size of the database in Gigabytes (GB)**.
+1. Click **Apply**.
+
+!!!note
+    Refer to [Sizing a SystemLink server](https://www.ni.com/documentation/en/systemlink/latest/setup/sizing-a-systemlink-server/) for guidance on the CPU, memory, and disk needed for SystemLink single node deployments.
+
+## Multi Node Deployments
+
+SystemLink supports three types of multi-node configuration. You can configure SystemLink to connect to a standalone instance, replica sets, or MongoDB Atlas.
+
+NI recommends connecting SystemLink to replica set of three or more MongoDB servers for all production deployments.
+
+!!!important
+    Due to a bug in a 3rd party MongoDB driver, SystemLink cannot connect to MongoDB instances where the MongoDB username provided in the form input or connection string contains either the `-` or the `_` character.
+
+- NI does not support configurations where multiple SystemLink servers use the same MongoDB instance.
+- NI does not support configurations where another production application uses the same MongoDB instance as SystemLink.
+
+!!!note
+    While SystemLink can connect to and use a sharded MongoDB cluster (`mongos`), it will not take advantage of horizontal scaling capabilities enabled by sharded clusters.
+
+When using a stand-alone MongoDB instance, replica set, or Atlas, you have the option to encrypt the data stored within the database. This is built into MongoDB Atlas or requires [MongoDB Enterprise Advanced](https://www.mongodb.com/products/mongodb-enterprise-advanced) for stand-alone or replica sets.
+
+!!!important "Connection String Formats"
+    Some connection string formats are unsupported or could cause issues in some environments.
+
+    - Connection strings that contain the query parameter `tls=` are unsupported. Use the `ssl=` query string to achieve the same capability with the same degree of security.
+    - In environments where the 4.2.1.1 root DNS server cannot be reached, the DNS seed list URI format (`mongodb+srv://`) will fail due to a bug ([Unable to connect to Atlas due to DNS connectivity issues #358](https://github.com/kobil-systems/mongodb/issues/358)) in a 3rd party MongoDB driver used by SystemLink. In this case use the [MongoDB standard connection string format](https://docs.mongodb.com/manual/reference/connection-string/#std-label-connections-standard-connection-string-format) to connect your replica set. This affects both self-hosted replica sets and MongoDB Atlas.
```

Comment:
The caveat to avoiding the `mongodb+srv://` url is that if you scale up / down the number of replicas, you need to update the connection string in the SystemLink configuration, which will require restarting services.  (Although it may require restarting services even with `mongodb+srv` urls - is the dns resolution done only once upfront? I have not tried this.)

Reply:
I've worked this point in as a sub-bullet about 4.2.1.1.